src/receiver.rs

```diff
@@ -69,22 +71,20 @@ impl Receiver {
                 header.key_id()
             );
 
-            let payload_begin_idx = skip + header.size();
-            let mut io_buffer: Vec<u8> = encrypted_frame[..skip]
-                .iter()
-                .chain(encrypted_frame[payload_begin_idx..].iter())
-                .copied()
-                .collect();
+            let payload_begin = skip + header.size();
+            self.buffer.clear();
+            self.buffer.extend(encrypted_frame[..skip].iter());
+            self.buffer.extend(encrypted_frame[payload_begin..].iter());
```

Comment:
might be that a `self.buffer.extend(&encrypted_frame[[payload_begin..]);` already would work since it implements `IntoIterator`. Haven't tried that though.

Reply:
I tried and it works, looks nicer too

```rust
            self.buffer.extend(&encrypted_frame[..skip]);
            self.buffer.extend(&encrypted_frame[payload_begin..]);
```

Reply:
Nice :+1: